runtime/moonbase/src/lib.rs

```diff
@@ -411,13 +411,18 @@ impl FeeCalculator for FixedGasPrice {
 		//       scaled by the multiplier, which means its multiplier will be overstated when
 		//       applied to an ethereum transaction
 		// note: transaction-payment uses both a congestion modifier (next_fee_multiplier, which is
-		//       updated once per block in on_initialize) and a 'WeightToFee' implementation. Our
+		//       updated once per block in on_finalize) and a 'WeightToFee' implementation. Our
 		//       runtime implements this as a 'ConstantModifier', so we can get away with a simple
 		//       multiplication here.
-		let total_weight_multiplier =
-			TransactionPayment::next_fee_multiplier().saturating_mul_int(currency::WEIGHT_FEE);
-		let gas_price = total_weight_multiplier * WEIGHT_PER_GAS as u128;
-		(gas_price.into(), 0u64.into())
+		// It is imperative that `saturating_mul_int` be performed as late as possible in the
+		// expression since it involves fixed point arithmetic with division, and will lead to
```

Comment:
This is well stated, but I think it could still call explicit attention to the "surprise" that `FixedU128` mul implicitly involves division and can truncate, e.g. "Note that `FixedU128`'s mul/arithmetic operations divide by a fixed divisor, which can silently introduce truncation."

That said, we're approaching a wall of text in this humble fn :sweat_smile: